test/test_jit.py

```diff
@@ -1259,6 +1259,30 @@ def fn(x, t):
         inputs = (torch.randn(1), (torch.randn(1), torch.randn(1)))
         self.checkTrace(fn, inputs)
 
+    def test_input_dict_flattens(self):
+        class Test(torch.nn.Module):
+            def forward(self, d):
+                return d['x'] + d['y']
+
+        inputs = {'x': torch.rand(3, 4), 'y': torch.rand(3, 4)}
+        module = torch.jit.trace(Test(), inputs)
+        kinds = [node.kind() for node in module.graph.nodes()]
+        self.assertEqual(kinds[0], 'aten::values')
+        self.assertEqual(kinds[1], 'prim::ListUnpack')
+
+    def test_input_dict_flattens_recursive(self):
+        class Test(torch.nn.Module):
+            def forward(self, d):
+                a, b = d['x']
+                return a + b
+
+        inputs = {'x': (torch.rand(2, 2), torch.rand(2, 2))}
+        module = torch.jit.trace(Test(), inputs)
+        kinds = [node.kind() for node in module.graph.nodes()]
+        self.assertEqual(kinds[0], 'aten::values')
+        self.assertEqual(kinds[1], 'prim::ListUnpack')
+        self.assertEqual(kinds[2], 'prim::TupleUnpack')
+
```

Comment:
lets add some tests that go through `checkTrace` and do various operations on the tensors in the dicts (mutation, pure ops, etc.)

Reply:
You mean the `checkTrace` in `test_jit.py`? Sure, I can do that. `checkTrace` itself will have to be modified, as it makes heavy assumptions about trace input types.